### ISSUE TITLE
Streaming

### ISSUE THREAD

**Issue by hieulbp** [moneytoo/Player#11]
Can u support play streaming like .m3u, m3u8, dash, hls,... tks very much

**Comment by moneytoo** [comment]
I was thinking about that but:

- I personally don't use it (and don't know much content to test it with).
- There are already other ExoPlayer based apps that are made (primarily or only) for streaming content.
- Wouldn't it also require additional UI? Bookmarked/recent streams?
- Supporting streaming would require the Internet permission. I'm actually perceiving not having permission to access Internet as an advantage - especially for the FOSS folks. I have already noticed [positive feedback](https://github.com/TeamNewPipe/NewPipe/issues/4175#issuecomment-748421713) regarding no internet permission.

**Comment by moneytoo** [comment]
On the other hand I understand there might be a lot of demand for this feature. I would also like to do that if that's just a few lines of code, especially since it doesn't need new action in control bar. To still offer offline version, I could just create another build variant, just like _Keepass2Android_ does.

**Comment by moneytoo** [comment]
I'm trying out streaming with this build: [Just.Player.v0.13-streaming.apk.zip](https://github.com/moneytoo/Player/files/5798849/Just.Player.v0.13-streaming.apk.zip)

Does this handle streaming as you would expect? I tested it with these file extensions: https://github.com/moneytoo/Player/blob/15bd89dbd3778f1e7ecc148d8632a703939c4770/app/src/main/AndroidManifest.xml#L56-L63

Is anything missing?

**Comment by hieulbp** [comment]
streaming is play url, but I could not more url and referer header

**Comment by CorvetteCole** [comment]
would love to get streaming support as well, will test shortly. This is the only app besides the default google video player that can play 4k60FPS on my chromecast with google tv

**Comment by moneytoo** [comment]
Basic support should be available in the `-online` version in [releases here on Github](https://github.com/moneytoo/Player/releases/latest). F-Droid and Google Play will keep the `-offline` build.

I only added support for extensions I was able to find and test so let me know in a new issue if anything is missing (with some sample to test).

@hieulbp I'm afraid that setting things like referer headers is too complex for this little app.
